File: Singly_Circluar_LinkedList/LinkedList.py
class Node :
    data = 0;
    next = None;

    def __init__(self) :
        self.data = 0;
        self.next = None;
    
class SinglyCL :
    __Head = None;
    __Tail = None;

    def __init__(self) :
        self.__Head = None;
        self.__Tail = None;
    
    def InsertFirst(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;

        if((self.__Head == None)and(self.__Tail == None)) :
            self.__Head = newn;
            self.__Tail = newn;
        else :
            newn.next = self.__Head;
            self.__Head = newn;
        (self.__Tail).next = self.__Head;

    def InsertLast(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;

        if((self.__Head == None)and(self.__Tail == None)) :
            self.__Head = newn;
            self.__Tail = newn;
        else :
            (self.__Tail).next = newn;
            self.__Tail = newn;
        (self.__Tail).next = (self.__Head).next;

    def InsertAtPos(self,value,ipos) :
        isize = 0;
        i = 0;
        isize = self.Count();
        newn = None;
        temp = self.__Head;

        if((ipos < 1)or(ipos > (isize + 1))) :
            return;
        elif(ipos == 1) :
            self.InsertFirst(value);
        elif(ipos == (isize + 1)) :
            self.InsertLast(value);
        else :
            newn = Node();
            newn.data = value;
            newn.next = None;

            for i in range(1,ipos - 1) :
                temp = temp.next;
            
            newn.next = temp.next;
            temp.next = newn;
    
    def DeleteFirst(self) :
        if((self.__Head == None)and(self.__Tail == None)) :   #If Linked List is Empty 
            return;
        elif(self.__Head == self.__Tail) :       #If LinkedLinked List Contains Only One Element
            del(self.__Head);
            self.__Head = None;
            self.__Tail = None;
        else :
            (self.__Head) = (self.__Head).next;
            del(self.__Tail).next;
            (self.__Tail).next = self.__Head;
    
    def DeleteLast(self) :
        temp = self.__Head;

        if((self.__Head == None)and(self.__Tail == None)) :
            return;
        elif(self.__Head == self.__Tail) :
            del(self.__Head);
            self.__Head = None;
            self.__Tail = None;
        else :
            while(temp.next != self.__Tail) :
                temp = temp.next;

            del(self.__Tail);
            self.__Tail = temp;
            (self.__Tail).next = self.__Head;
        
    def DeleteAtPos(self,ipos) :
        i = 0;
        isize = 0;
        temp = self.__Head;
        temp2 = None;

        isize = self.Count();

        if((ipos < 1)or(ipos > isize)) :
            return;
        elif(ipos == 1) :
            self.DeleteFirst();
        elif(ipos == isize) :
            self.DeleteLast();
        else :
            for i in range(1,ipos - 1) :
                temp = temp.next;
            
            temp2 = temp.next;
            temp.next = temp2.next;
            del(temp2);
            temp2 = None;

    def Display(self) :
        temp = self.__Head;

        if((self.__Head == None)and(self.__Tail == None)) :
            return;
        else :
            print("Elements of Linked List are : ");
            while(temp != self.__Tail) :
                print("|%d|<->"%temp.data,end = "");
                temp = temp.next;
            print("|%d|<->"%temp.data,end = "");
            print();
            
    def Count(self) :
        temp = self.__Head;
        icnt = 0;

        if((self.__Head == None)and(self.__Tail == None)) :
            return 0;
        else :
            while(temp != self.__Tail) :
                icnt = icnt + 1;
                temp = temp.next;
            icnt = icnt + 1;
            
            return icnt;

def main() :
    print("Jay Ganesh.....");
    iret = 0;
    ino = 0;
    ielem = 0;

    sobj = SinglyCL();
    print("--------------------------------Singly Circular Linked List Demonstration-----------------------------------");
    while(True) :
        print("Select Any Operaton From Below List of Operations : ");
        print("1. Insert Element At First Position.");
        print("2. Insert Element At Last Position.");
        print("3. Insert Element At Given Position.");
        print("4. Display Elements of Linked List.");
        print("5. Count Elements Present in Linked List.");
        print("6. Delete Element From First Position.");
        print("7. Delete Element From Last Position.");
        print("8. Delete Element From Given Position.");
        print("0. Exit.");

        print("Enter Your Choice : ");
        ino = int(input());

        if(ino == 1) :
            print("Enter Value To Insert : ");
            ielem = int(input());

            sobj.InsertFirst(ielem);
        elif(ino == 2) :
            print("Enter Value To Insert : ");
            ielem = int(input());

            sobj.InsertLast(ielem);
        elif(ino == 3) :
            print("Enter Value To Insert : ");
            ielem = int(input());

            print("Enter Position You Want To Insert Element At : ");
            ipos = int(input());

            sobj.InsertAtPos(ielem,ipos);
        elif(ino == 4) :
            sobj.Display();
        elif(ino == 5) :
            iret = sobj.Count();
            print("Count of Elements Present in Linked List is : ",iret);
        elif(ino == 6) :
            sobj.DeleteFirst();
        elif(ino == 7) :
            sobj.DeleteLast();
        elif(ino == 8) :
            print("Enter Position You Want To Delete Element From : ");
            ipos = int(input());

            sobj.DeleteAtPos(ipos);
        elif(ino == 0) :
            return;
        else :
            print("Invalid Option Number Entered, Please Enter Valid Option");
            return;

if __name__ == "__main__" :
    main();

File: Singly_Circluar_LinkedList/LinkedList.java
import java.lang.*;
import java.util.*;

class Node 
{
    int data;
    Node next;

    Node()
    {
        this.data = 0;
        this.next = null;
    }
}

class SinglyCL 
{
    private Node Head;
    private Node Tail;

    SinglyCL()
    {
        this.Head = null;
        this.Tail = null;
    }

    void InsertFirst(int value)
    {
        Node newn = new Node();
        newn.data = value;
        newn.next = null;

        if((this.Head == null)&&(this.Tail == null))
        {
            this.Head = newn;
            this.Tail = newn;
        }
        else 
        {
            newn.next = this.Head;
            this.Head = newn;
        }
        (this.Tail).next = this.Head;
    }

    void InsertLast(int value)
    {
        Node newn = new Node();
        newn.data = value;
        newn.next = null;

        if((this.Head == null)&&(this.Tail == null))
        {
            this.Head = newn;
            this.Tail = newn;
        }
        else 
        {
            (this.Tail).next = newn;
            this.Tail = newn;
        }
        (this.Tail).next = this.Head;
    }

    void InsertAtPos(int value,int ipos)
    {
        int i = 0;
        int isize = 0;
        Node temp = this.Head;
        Node newn = null;

        isize = this.Count();

        if((ipos < 1)||(ipos > (isize + 1)))
        {
            return;
        }
        else if(ipos == 1)
        {
            this.InsertFirst(value);
        }
        else if(ipos == (isize + 1))
        {
            this.InsertLast(value);
        }
        else 
        {
            newn = new Node();
            newn.data = value;
            newn.next = null;

            for(i = 1;i < (ipos - 1);i++)
            {
                temp = temp.next;
            }

            newn.next = temp.next;
            temp.next = newn;
        }
    }

    void DeleteFirst()
    {
        if((this.Head == null)&&(this.Tail == null))
        {
            return;
        }
        else if(this.Head == this.Tail)
        {
            this.Head = null;
            this.Tail = null;

            return;
        }
        else 
        {
            this.Head = (this.Head).next;
            this.Tail.next = null;
            this.Tail.next = this.Head;
        }
    }

    void DeleteLast()
    {
        Node temp = this.Head;

        if((this.Head == null)&&(this.Tail == null))
        {
            return;
        }
        else if(this.Head == this.Tail)
        {
            this.Head = null;
            this.Tail = null;

            return;
        }
        else 
        {
            while(temp.next != this.Tail)
            {
                temp = temp.next;
            }

            temp.next = null;
            this.Tail = temp;
            temp.next = this.Head; 
        }
    }

    void DeleteAtPos(int ipos)
    {
        int isize = 0;
        int i = 0;
        Node temp = this.Head;
        Node temp2 = null;

        isize = this.Count();

        if((ipos < 1)||(ipos > isize))
        {
            return;
        }
        else if(ipos == 1)
        {
            this.DeleteFirst();
        }
        else if(ipos == isize)
        {
            this.DeleteLast();
        }
        else 
        {
            for(i = 1;i < (ipos - 1);i++)
            {
                temp = temp.next;
            }

            temp2 = temp.next;
            temp.next = temp2.next;

            temp2 = null;
        }
    }

    void Display()
    {
        Node temp = null;
        temp = this.Head;

        if((this.Head == null)&&(this.Tail == null))
        {
            return;
        }
        else 
        {

            System.out.println("Elements of Linked List are : ");
            do
            {
                System.out.printf("|%d|<->",temp.data);
                temp = temp.next;
            }while(temp != (this.Tail).next);        
            System.out.println();
        }
    }

    int Count()
    {
        Node temp = null;
        temp = this.Head;
        int icnt = 0;

        if((this.Head == null)&&(this.Tail == null))
        {
            return 0;
        }
        else 
        {
            
            do
            {
                icnt++;
                temp = temp.next;
            }while(temp != (this.Tail).next);        

            return icnt;
        }
    }
}

class LinkedList 
{
    public static void main(String args[])
    {
        int iret = 0;
        int ielem = 0;
        int ino = 0;
        int ipos = 0;

        System.out.println("Jay Ganesh.....");

        System.out.println("-------------Singly Cicular Linked List Demonstration(Java)------------------");
        
        Scanner sobj = new Scanner(System.in);

        SinglyCL scobj = new SinglyCL();

        while(true)
        {
            System.out.println("Select Any Operation From Below List of Operations To Perform on Linked List : ");
            System.out.println("1. Insert Element At First Position.");
            System.out.println("2. Insert Element At Last Position.");
            System.out.println("3. Insert Element At Entered Position.");
            System.out.println("4. Display Elements of Linked List.");
            System.out.println("5. Count No of Elements Present in Linked List.");
            System.out.println("6. Delete Element From First Position.");
            System.out.println("7. Delete Element From Last Position.");
            System.out.println("8. Delete Element From Entered Position.");
            System.out.println("0. Exit.");

            System.out.println("Enter Option Number : ");
            ino = sobj.nextInt();

            switch(ino)
            {
                case 1 :
                    {
                        System.out.println("Enter Number to Insert : ");
                        ielem = sobj.nextInt();

                        scobj.InsertFirst(ielem);

                        break;
                    }
                case 2 :
                    {
                        System.out.println("Enter Number to Insert : ");
                        ielem = sobj.nextInt();

                        scobj.InsertLast(ielem);
                        
                        break;
                    }
                case 3 :
                    {
                        System.out.println("Enter Number to Insert : ");
                        ielem = sobj.nextInt();


                        System.out.println("Enter Position to Insert Element At : ");
                        ipos = sobj.nextInt();

                        scobj.InsertAtPos(ielem,ipos);

                        break;
                    }
                case 4 :
                    {
                        scobj.Display();
                        
                        break;
                    }
                case 5 :
                    {
                        iret = scobj.Count();
                        
                        System.out.printf("Count of Elements is : %d\n",iret);
                        break;
                    }
                case 6 :
                    {   
                        scobj.DeleteFirst();
                        break;
                    }
                case 7 :
                    {   
                        scobj.DeleteLast();
                        break;
                    }
                case 8 :
                    {
                        System.out.println("Enter Position to Delete Element From : ");
                        ipos = sobj.nextInt();

                        scobj.DeleteAtPos(ipos);
                        
                        break;
                    }
                case 0 :
                    {
                        return;
                    }
                default :
                {
                    System.out.println("Invalid Option Number Entered, Please Enter Valid Option Number.");
                    return;
                }
            }           
        }
    }
}